modules/vmfcore/java/com/intel/vmf/Variant.java

```diff
@@ -0,0 +1,207 @@
+package com.intel.vmf;
+
+import java.nio.ByteBuffer;
+
+ public class Variant
+ {
+    static
+    {
+        System.loadLibrary("vmf");
+    }
+	
+    public static final int type_unknown = 0;
+    public static final int type_char = 1;
+    public static final int type_integer = 2;
+    public static final int type_real = 3;
+    public static final int type_string = 4;
+    public static final int type_vec2d = 5;
+    public static final int type_vec3d = 6;
+    public static final int type_vec4d = 7;
+    public static final int type_rawbuffer = 8;
+    public static final int type_char_vector = 9;
+    public static final int type_integer_vector = 10;
+    public static final int type_real_vector = 11;
+    public static final int type_string_vector = 12;
+    public static final int type_vec2d_vector = 13;
+    public static final int type_vec3d_vector = 14;
+    public static final int type_vec4d_vector = 15;
+    
+    protected final long nativeObj;
+	
+    protected Variant (long addr)
+    {
+        if (addr == 0)
+            throw new java.lang.UnsupportedOperationException("Native object address is NULL");
+        
+        nativeObj = addr;
+    }
+    
+    public Variant ()
+    {
+        nativeObj = n_Variant ();
+    }
+	
+    public Variant (int value)
```

Comment:
should be `long value`, but we need to avoid conflict with the protected c-tor...